jsk_interactive_markers/jsk_interactive_marker/src/marker_6dof.cpp

```diff
@@ -242,7 +244,9 @@ class Marker6DOF {
 
   void publishTF(const geometry_msgs::PoseStamped& pose) {
     tf::Transform transform;
-    tf::poseMsgToTF(pose.pose, transform);
+    geometry_msgs::PoseStamped transformed_pose;
+    tf_listener_->transformPose(frame_id_, ros::Time(0), pose, pose.header.frame_id, transformed_pose);
+    tf::poseMsgToTF(transformed_pose.pose, transform);
```

Comment:
The main use case of this method is below,
```c++
  void timerTFCallback(const ros::TimerEvent& e) {
    boost::mutex::scoped_lock lock(mutex_);
    geometry_msgs::PoseStamped pose = latest_pose_;
    pose.header.stamp = e.current_real;
    publishTF(pose);
  }
```
and `latest_pose_.header.frame_id` is same as `frame_id_`,
so  `frame_id_` and `pose.header.frame_id` is same.
In that case, in what situation the `pose` should be updated with looking up transform?